packages/plugin-json/src/lib/validation.ts

```diff
@@ -39,14 +39,39 @@ const isNonNullLocalizedField = (field: FieldDefinitionNode) =>
 export const getTooManyFiltersReport = (types: ObjectTypeDefinitionNode[]) =>
   getFieldsReport(
     types.filter(
-      (type) => (type.fields?.filter(filterableField).length ?? 0) > 5
+      (type) => (type.fields?.filter(filterableField)?.length ?? 0) > 5
     ),
     filterableField
   );
 
 const filterableField = (field: FieldDefinitionNode) =>
   hasDirective(field, "amplienceFilterable");
 
+export const getTooManyDeliveryKeysReport = (types: ObjectTypeDefinitionNode[]) =>
+  getFieldsReport(
+    types.filter(
+      (type) => (type.fields?.filter(deliveryKeyField)?.length ?? 0) > 1
+    ),
+    deliveryKeyField
+  )
+
+export const getDeliveryKeyNotNullableStringReport = (types: ObjectTypeDefinitionNode[]) =>
+  getFieldsReport(
+    types.filter(
+      (type) => type.fields?.some((field) => deliveryKeyField(field) && !nullableStringField(field))
+    ),
+    (field) => deliveryKeyField(field) && !nullableStringField(field)
+  )
+
+const deliveryKeyField = (field: FieldDefinitionNode) =>
```

Comment:
Please rename this to `isDeliveryKeyField`

Reply:
Done (as well as for `nullableStringField`). I had named them this way to be consistent with the existing `filterableField`, so I renamed that to `isFilterableField` to match.